src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedToArray = sourceString.split(';');
+
+  return convertedToArray.reduce((obj, property) => {
+    const style = property.split(':');
+
+    let [key, value] = style;
+
+    key = key.trim();
+    value = value ? value.trim() : '';
+
+    if (value !== '') {
+      return {
+        ...obj,
+        [key]: value,
+      };
+    }
+
+    return obj;
```

Comment:
The reduce function is creating a new object on each iteration which is unnecessary and could lead to performance issues for a large number of CSS declarations. Consider using the object that is passed as the accumulator (`obj`) directly, without spreading it into a new object.